tools/setup/app.nim

```diff
@@ -0,0 +1,138 @@
+import std/os
+import std/osproc
+import std/httpclient
+import std/strutils
+import pkg/chronicles
+import pkg/questionable
+import pkg/ethers
+import ./print
+import ./networkconfig
+
+type App* = ref object
+  configLines: seq[string]
+  networkConfig: ?ArchivistNetwork
+
+  ethAddress: string
+  # if we have networkConfig AND storage mode (not manual) is selected
+  # then run CIRDL
+  storageModeSelected*: bool
+  # if testnet or devnet, display faucet links
+  faucetLinkNetwork*: string
+  # if set, display webui link
+  webUi*: bool
+
+proc writeConfigLine*(app: App, line: string) =
+  app.configLines.add(line)
+
+proc fetchNetworkConfig*(app: App, network: string): ArchivistNetwork =
+  if isNone(app.networkConfig):
+    info "Fetching network information...", network
+    app.networkConfig = some getNetworkConfig(network)
+  return !app.networkConfig
+
+proc fetchPublicIp*(app: App): string =
+  let
+    url = "http://ip.archivist.storage"
+    client = newHttpClient()
+  try:
+    info "Fetching public IP...", url
+    return client.getContent(url)
+  finally:
+    client.close()
+
+proc saveFile(filename: string, content: string) =
+  if fileExists(filename):
+    removeFile(filename)
+
+  let f = open(filename, fmWrite)
+  f.writeLine(content)
+  f.close()
+
+proc createNewEthKeyfile*(app: App, privKeyFilename: string, addressFilename: string) =
+  info "Generating Ethereum wallet...", privKeyFilename, addressFilename
+  var rng = keys.newRng()[]
+  let
+    wallet = Wallet.createRandom()
+    keyStr = "0x" & $(wallet.privateKey)
+  app.ethAddress = $(wallet.address)
+
+  saveFile(privKeyFilename, keyStr)
+  saveFile(addressFilename, app.ethAddress)
+
+proc findCirdl(): string =
+  try:
+    for file in walkDir(".", true):
```

Comment:
Doesn't this depend on the working directory when you invoke the setup script? Perhaps you can look relative to the path of the setup executable?